_posts/2020-03-02-ios-deploys-super-weird-totally-normal.markdown

```diff
@@ -0,0 +1,113 @@
+deploy, fix the bug, and re-deploy with the fix.
+
+(Note: some readers might be wondering why we don't take advantage of over-the-air updates to our React Native
+JavaScript bundle. This is definitely possible, but our app is brownfield with some native code and some React
+Native code and we haven't yet built out the infrastructure for this. As more and more code shifts to React Native,
+we plan to investigate OTA updates to JavaScript bundles.)
+
+The second reason to deploy iOS software on a schedule, rather than with continuous delivery, depends on the App
+Store review process. This is another big difference that takes web engineers a while to get used to. Whenever we
+ship a version of our iOS app, we ship it to Apple for review, and then Apple ships it to our users. I'm not going
+to debate the utility of Apple as an intermediary – the fact is that they own the iOS platform and these are the
+rules they have chosen. If you want to ship iOS software, then you have to abide by Apple's rules.
+
+App Store review isn't exactly QA. I mean, if we were to ship an app update and it crashes upon launch, Apple isn't
+likely to approve that update. But if we have a small bug buried somewhere in the app, we can't expect Apple to
+find it. Apple is only testing for adherence to their
+[App Store Review Guidelines](https://developer.apple.com/app-store/review/guidelines/). Apple is looking for
+things like: is the app trying to steal user data? Is the app displaying objectionable material, given its age
+rating? Is the app's description and App Store metadata correct? That kind of stuff.
+
+So not only do iOS software developers need a lot of confidence in every deploy, but they also need to abide by
+Apple's guidelines.
+
+Alright. Let's revisit the situation from earlier.
+
+We had a bug fix in our app, and getting it deployed was blocking an important change to our back-end API. First,
+we had to consider that some users simply wouldn't get the update. We had to ask ourselves if this would block the
+back-end change entirely – through open discussions with the team, we decided to move forward. Second, we had to
+consider the other work that had been merged since our last release. Were we confident in deploying that work
+as-is? And what amount of QA would need to be done to ship those changes as well?
+
+That last point is really interesting because there was actually another option. Rather than deploy the app based
+off the current `master` branch (with the bug fix _and_ other work included), we did something a bit clever.
+[fastlane](https://fastlane.tools), the tool we use to automate our iOS deploys, will tag each commit that we
+submit to the App Store. So rather than deploy the current `master` branch, which would require very rigorous QA,
+we checked out the previous release tag. We then used
+[`git cherry-pick`](https://www.atlassian.com/git/tutorials/cherry-pick) to apply _only_ the bug fix changes, and
+deployed from there.
```

Comment:
Any screenshots or code examples that could be easily incorporated here? Something showing `cherry-pick` in action would be a nice visual

Reply:
I'm avoiding linking to the code explicitly, and `cherry-pick` is involved enough that a single visual would distract more than help (which is why I link to further reading). I've looked for Slack threads to provide context, but I'd rather not detail the nature of our backend change.

Reply:
Ah, I did find a screenshot that made sense, thanks!